src/algebra/regular.lean

```diff
@@ -57,8 +57,8 @@ lemma of_left_right {a : R} (hl : is_left_regular a) (hr : is_right_regular a) :
 ⟨hl, hr⟩
 
 /-- An element admitting a left and a right inverse is regular. -/
-lemma of_mul_eq_one_mul_eq_one {a ai : R} (hr : a * ai = 1) (hl : ai * a = 1) : is_regular a :=
-of_left_right (left_of_mul_eq_one hl) (right_of_mul_eq_one hr)
+lemma unit.is_regular (a : unit R) : is_regular (a : R) :=
```

Comment:
Whoops, I meant `units`

Reply:
But then it's a special case of the lemma about cancellative monoids.

Reply:
So, should I roll back to what it was or not state it?  My main reason for stating it was to prove `is_unit.is_regular`.

Reply:
Is it? It has `coe a` instead of `a` in the conclusion.

Reply:
Eric, in the end, your version made it through: the coercion appears to be the deciding factor!

Discussion here:

https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/is_regular